eng/common/scripts/release-docs-metadata.ps1

```diff
@@ -0,0 +1,208 @@
+# Note, due to how `Expand-Archive` is leveraged in this script,
+# powershell core is a requirement for successful execution.
+param (
+  # arguments leveraged to parse and identify artifacts
+  $ArtifactLocation, # the root of the artifact folder. DevOps $(System.ArtifactsDirectory)
+  $WorkDirectory, # a clean folder that we can work in
+  $ReleaseSHA, # the SHA for the artifacts. DevOps: $(Release.Artifacts.<artifactAlias>.SourceVersion) or $(Build.SourceVersion)
+  $RepoId, # full repo id. EG azure/azure-sdk-for-net  DevOps: $(Build.Repository.Id). Used as a part of VerifyPackages
+  $Repository, # EG: "Maven", "PyPI", "NPM"
+
+  # arguments necessary to power the docs release
+  $Language, # EG: js, java, dotnet. Used in language for the embedded readme.
+  $TargetDocRepo, # the repository that we will be writing the readmes to 
+  $LatestBranch = "smoke-test", # normally `smoke-test`
+  $PreviewBranch = "smoke-test-preview", # normally `smoke-test-preview`
+  $DocRepoContentLocation = "docs-ref-services/" # within the doc repo, where does our readme go?
+)
+
+# import artifact parsing and semver handling
+. (Join-Path $PSScriptRoot artifact-metadata-parsing.ps1)
+. (Join-Path $PSScriptRoot SemVer.ps1)
+
+# normally best case is to stay away from `Invoke-Expression`
+# however, given that we have a few git commands to get through and simplicity of script has a value of it's own
+# this works in this limited context.
+function _execute($command, $hardErr = $True){
+  try {
+    Write-Host $command
+    Invoke-Expression "&$command"
+    if ($LastExitCode -ne 0) {
+      Write-Host "Command exited with error code $LastExitCode"
+    }
+  }
+  catch {
+    WriteHost $_
+    if ($hardErr) {
+      exit(1)
+    }
+  }
+}
+
+function SyncDocRepo($docRepo, $workingDirectory, $sourceBranch, $prBranch) {
+  # get the doc repo, looking for the rdm branch if it exists
+  $docRepo = "https://github.com/$docRepo"
+  $repoRoot = (Join-Path $workingDirectory "repo/" )
+
+  _execute("git clone $docRepo $repoRoot")
```

Comment:
This function looks to be doing more then we probably need it to do. I assume all you want to do is clone the repo and checkout a particular branch, if it exists otherwise create it.  Similar to my comments in my other file I don't expect that you need to actually check for the branch, you simply check it out and it will either track the upstream version if it exists and if not it will create a new branch from the clone point.

Reply:
Might be worth having a look at https://github.com/Azure/azure-sdk-tools/blob/master/eng/pipelines/templates/steps/sync-directory.yml which is what we use to sync the eng/common directory.

Reply:
I _don't_ want to only checkout a particular branch. I want to accumulate readme changes to a _specific_ branch. I don't particularly want to file a new PR every time. See where I'm coming from?

Reply:
You don't need a new PR for each one and you can totally create a commit per change if you like however if you are making multiple commits from the same clone you will definitely hit the branch creation problem you were seeing. 

Is there a reason you want multiple commits from the same set of readme updates? Won't they be squashed on merged? If we do feel that is an interesting scenario I think we can come up with some options. I'll think about that a scenario a little more.

Reply:
Refactor is pushed for a branch per readme. Gotta test the params in a test release though.